### PR TITLE
fix(google-maps): avoid runtime error on server 

### PR DESCRIPTION
**Pull request by devversion**
* See individual commits.

Caretaker: Please preserve commits when merging this.

### CODE REVIEW

File: src/google-maps/map-info-window/map-info-window.ts [hunk 1]
```diff
@@ -122,7 +122,12 @@ export class MapInfoWindow implements OnInit, OnDestroy {
     this._eventManager.destroy();
     this._destroy.next();
     this._destroy.complete();
-    this.close();
+
+    // No info window has been created on the server. To avoid errors on the server,
+    // we do not call `close` here as it expects an instantiated info window instance.
+    if (this._googleMap._isBrowser) {
```
Comment (crisbeto) [inline src/google-maps/map-info-window/map-info-window.ts:128]: It might be better to check if the `this._infoWindow` is defined.

Comment (devversion) [reply src/google-maps/map-info-window/map-info-window.ts:128]: I think I should just expand the comment to mention that also there is no google-map instance, so also `_assertInitialized` in `close` will fail.

Comment (crisbeto) [reply src/google-maps/map-info-window/map-info-window.ts:128]: That's depending on information from outside the class though. If we check for the `_infoWindow`, we'd only be depending on the local state of the component.

Comment (devversion) [reply src/google-maps/map-info-window/map-info-window.ts]: I would personally still prefer `isBrowser` as `infoWindow` requires context on what the `close` method does. I don't feel strong about it, so updated to `_infoWindow`.

**Pull request merged**